No auto completion with `muhammed-sammy.csharp`

With Onivim 2 `0.5.9-nightly | 0cd8c424d` on Arch I cannot get auto completion to work on any C# file using the extension `muhammed-sammy.csharp`.

When attempting to open the auto completion box this message appears:

```
[ERROR] [30.617s] Exthost.Client : Request 52 for $provideCompletionItems failed with error: ParseFailedException("Expected null or:\n  in field \"b\":\n    while decoding a list:\n      element 0:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"AccessViolationException\", \"description\": \"\" }\n      element 1:\n        in field \"a\":\n          Expected a string, but got { \"label\": \"Action\", \"description\": \"\" }\n      element 2:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"Action<>\", \"description\": \"\" }\n      element 3:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"Activator\", \"description\": \"\" }\n      element 4:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"AggregateException\", \"description\": \"\" }\n      (...431 errors not shown...)")
[ERROR] [30.636s] Exthost.Client : Request 48 for $provideCompletionItems failed with error: ParseFailedException("Expected null or:\n  in field \"b\":\n    while decoding a list:\n      element 0:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"AccessViolationException\", \"description\": \"\" }\n      element 1:\n        in field \"a\":\n          Expected a string, but got { \"label\": \"Action\", \"description\": \"\" }\n      element 2:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"Action<>\", \"description\": \"\" }\n      element 3:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"Activator\", \"description\": \"\" }\n      element 4:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"AggregateException\", \"description\": \"\" }\n      (...431 errors not shown...)")
[ERROR] [33.129s] Exthost.Client : Request 88 for $provideCompletionItems failed with error: ParseFailedException("Expected null or:\n  in field \"b\":\n    while decoding a list:\n      element 0:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"CompareTo\", \"description\": \"\" }\n      element 1:\n        in field \"a\":\n          Expected a string, but got { \"label\": \"Equals\", \"description\": \"\" }\n      element 2:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"GetHashCode\", \"description\": \"\" }\n      element 3:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"GetType\", \"description\": \"\" }\n      element 4:\n        in field \"a\":\n          Expected a string, but got\n          { \"label\": \"GetTypeCode\", \"description\": \"\" }\n      (...9 errors not shown...)")
```

--- a/CHANGES_CURRENT.md
+++ b/CHANGES_CURRENT.md
@@ -6,6 +6,7 @@
 ### Bug Fixes
 
 - #3791 - SCM: Fix 'Git: Checkout to' command (fixes #2525)
+- #3800 - Completion: Fix completion item parsing resiliency (fixes #3787)
 
 ### Performance
 

--- a/src/Exthost/SuggestItem.re
+++ b/src/Exthost/SuggestItem.re
@@ -128,13 +128,25 @@ let sortText = ({sortText, label, _}) => {
   };
 };
 
+module SuggestItemLabel = {
+  type t = string;
+
+  let decode =
+    Json.Decode.(
+      one_of([
+        ("string", string),
+        ("record", obj(({field, _}) => {field.required("label", string)})),
+      ])
+    );
+};
+
 module Dto = {
   let decode = (~defaultRange) => {
     Json.Decode.(
       obj(({field, _}) => {
         // These fields come from the `ISuggestDataDtoField` definition:
         // https://github.com/onivim/vscode-exthost/blob/50bef147f7bbd250015361a4e3cad3305f65bc27/src/vs/workbench/api/common/extHost.protocol.ts#L1089
-        let label = field.required("a", string);
+        let label = field.required("a", SuggestItemLabel.decode);
 
         let kind =
           field.withDefault(
@@ -147,11 +159,12 @@ module Dto = {
                ),
           );
 
-        let detail = field.optional("c", string);
-        let documentation = field.optional("d", MarkdownString.decode);
-        let sortText = field.optional("e", string);
-        let filterText = field.optional("f", string);
-        let insertText = field.optional("h", string);
+        let detail = field.withDefault("c", None, nullable(string));
+        let documentation =
+          field.withDefault("d", None, nullable(MarkdownString.decode));
+        let sortText = field.withDefault("e", None, nullable(string));
+        let filterText = field.withDefault("f", None, nullable(string));
+        let insertText = field.withDefault("h", None, nullable(string));
         let insertTextRules =
           field.withDefault(
             "i",

